File: src/1578. Minimum Time to Make Rope Colorful.py
class Solution:
    def minCost(self, colors: str, neededTime: List[int]) -> int:
        n = len(colors)
        ans = i = j = 0
        while i < n and j < n:
            max_time = neededTime[i]
            while j < n and colors[i] == colors[j]:
                ans += neededTime[j]
                max_time = max(max_time, neededTime[j])
                j += 1
            ans -= max_time
            i = j
        return ans

File: src/1578. Minimum Time to Make Rope Colorful.cpp
class Solution {
public:
    int minCost(string colors, vector<int>& neededTime) {
        int n = colors.size(),
            ans = 0,
            i = 0,
            j = 0;
        while (i < n && j < n) {
            int maxTime = neededTime[i];
            while (j < n && colors[i] == colors[j]) {
                ans += neededTime[j];
                maxTime = max(maxTime, neededTime[j]);
                j++;
            }
            ans -= maxTime;
            i = j;
        }
        return ans;
    }
};